src/main/kotlin/se/uulm/snowballr/backend/service/CriterionService.kt

```diff
@@ -76,35 +76,31 @@ class CriterionService(
      * @throws SnowballRException.FailedPreconditionException If attempting to access a criterion
      *         in a project that is not active.
      */
-    private suspend fun checkProjectCriterionPermission(
-        criterion: Criterion,
-        currentUser: User,
-        accessType: AccessType,
-    ) {
-        val criterionId = criterion.id.toString()
-        val project = projectRepo.getProjectById(criterion.projectId!!)
+    private suspend fun checkProjectCriterionPermission(projectId: UUID, currentUser: User, accessType: AccessType) {
+        val project = projectRepo.getProjectById(projectId)
         val members = when (accessType) {
             AccessType.READ -> projectMemberRepo.getMembersOfProject(project.id)
+            AccessType.CREATE -> projectMemberRepo.getAllProjectAdmins(project.id)
             AccessType.UPDATE -> projectMemberRepo.getAllProjectAdmins(project.id)
```

Comment:
We combine those two cases.

```suggestion
            AccessType.UPDATE, AccessType.CREATE -> projectMemberRepo.getAllProjectAdmins(project.id)
```